### ISSUE TITLE
CR10S PRO - Bltouch V3

### ISSUE THREAD

**Issue by mennovanhout** [InsanityAutomation/Marlin#32]
### Description

Just installed the latest firmware sadly enough my bltouch v3 still doesn't stop when he hits something.

### Steps to Reproduce

1. Install BLTouch
2. Update LCD and printer firmware
3. Zhome and hold your hand underneath the bltouch

**Expected behavior:** That the printer stops lowering the z axis

**Actual behavior:** Keeps lowering.

**Comment by chunter1** [comment]
I had the same problem.
I measured the ouput signal of my BLTouch and saw, that it never gets high.
The solution for me was to connect a 1kOhm resistor between the BLTouch output and 5V to pull the output high (it is an open drain output).
Maybe this issue can also be solved by enabling an uC internal pullup?
My board is a Creality3D v2.4 Motherboard with an Atmega2560. (CR-10s Pro).

**Comment by leseaw** [comment]
https://www.antclabs.com/wiring3-1

**Comment by chunter1** [comment]
@leseaw
Is this capacitor also labeled C7 on the CR-10s Pro motherboard?
Or is there even the schematics of the board available?

**Comment by mennovanhout** [comment]
> @leseaw
> Is this capacitor also labeled C7 on the CR-10s Pro motherboard?
> Or is there even the schematics of the board available?

No it is not. Also this page is about v1 mainboards howerver 99% of cr10s pro's have the V2.1 mainboard. there is 1 capacitor near the z- input and that is c48. But the question is will this fix it? I don't dare to try it.

**Comment by chunter1** [comment]
On my Creality3d v2.4 motherboard, the capacitor is labeled C48 and it's a 100 nF type.
Additionally there are a 10k pullup resistor and a 10k series resistor (i guess) going to the GPIO.

**Comment by mennovanhout** [comment]
> On my Creality3d v2.4 motherboard, the capacitor is labeled C48 and it's a 100 nF type.
> Additionally there are a 10k pullup resistor and a 10k series resistor (i guess) going to the GPIO.

I removed c48 yesterday. Didn't solve it. I'm going to add 1k resistors today.

**Comment by chunter1** [comment]
I will fit the 100nF capacitor again since it does not have a noticable effect on the measurements but increases noise rejection/false triggering and ESD protection.
Since i am using the additional 1k pullup (10k || 1k), i get a time constant of the RC circuit of 91 us.
To trigger the poitive logic level, it takes roughly 2 x Tau = 273 us which is perfectly fine.
In the opposite direction, the "relevant" one for measuring, it takes even less time since the internal resistance of the BLTouch output is much lower and discharges the capacitor way faster.
Later today, i will hook up my scope to verify the signal edges. :).

**Comment by mennovanhout** [comment]
> I will fit the 100nF capacitor again since it does not have a noticable effect on the measurements but increases noise rejection/false triggering and ESD protection.
> Since i am using the additional 1k pullup (10k || 1k), i get a time constant of the RC circuit of 91 us.
> To trigger the poitive logic level, it takes roughly 2 x Tau = 273 us which is perfectly fine.
> In the opposite direction, the "relevant" one for measuring, it takes even less time since the internal resistance of the BLTouch output is much lower and discharges the capacitor way faster.
> Later today, i will hook up my scope to verify the signal edges. :).

I can't solder good enough to reinstall the capicator :(.
But I'll add a 1k resistor today between the red and white wire and post my findings here.

**Comment by mennovanhout** [comment]
1k Resistor is not working for me. The BLtouch gets way to less power now. The blue light is almost out and the pin won't come out..... what now?

**Comment by chunter1** [comment]
I just measured the BLTouch output signal and it ranges from 0 to 2V only.
Don't know yet why it does not get pulled up to 5V...

**Comment by mennovanhout** [comment]
Okay so I got mine to work perfectly.
I removed the c48 capacitor and added a 1k transistor between the red and white wire. This boosts the white wire with 5v from the red wire which solves the problem @chunter1 .

I have to resolder this: But it was a quick test and works perfectly:

![60389720_310584139835354_3129861931247075328_n](https://user-images.githubusercontent.com/26040589/57722246-4a8d9a00-7686-11e9-95d1-cd44953c0d56.jpg)

**Comment by chunter1** [comment]
Fine to hear, but is your high-level on the signal line also limited to 2V only?

I am playing with these settings but see no change:

//#define BLTOUCH_DELAY 500   // (ms) Enable and increase if needed
//#define BLTOUCH_FORCE_5V_MODE // For BLTouch V3.0 force 5V only mode

**Comment by chunter1** [comment]
And this is what my rising and falling edges look like no matter what i change in the previously mentioned settings:
Don't overlook the different time scales.

![falling_edge](https://user-images.githubusercontent.com/16907942/57722959-deac3100-7687-11e9-8dbb-dd1db5f4e277.jpg)
![rising_edge](https://user-images.githubusercontent.com/16907942/57722960-df44c780-7687-11e9-94c5-5ea9d626bc8a.jpg)

**Comment by chunter1** [comment]
Very interesting thread:
https://github.com/MarlinFirmware/Marlin/issues/13345

**Comment by chunter1** [comment]
Not connecting the BLtouch output to the Creality V3.0 motherboard gives the correct 5V logic signal with clean edges. :/

**Comment by mennovanhout** [comment]
> Not connecting the BLtouch output to the Creality V3.0 motherboard gives the correct 5V logic signal with clean edges. :/

Probably the capicator that is eating your power. I've got a new problem i can't put the z offset lower then -0.52 ?? it jumps back ...

**Comment by chunter1** [comment]
There is definately a configuration issue because the Z-MIN input pin is pulling low.
As far as i have seen the signal is routed to GPIO PD3 of the Atmega2560.

**Comment by chunter1** [comment]
@mennovanhout 
ATTENTION!
Do not use a low resistor value!
I tried 330 Ohms to get reliable triggering and killed my BLTouch!
Luckily only the two schottky diodes were defect and i could replace them :)
So there is definately a problem with the configuration setup.
The GPIO seems to be pulling low and thus we hardly can get the minimum voltage level to sense the line as high!
Maybe some developer can help us with the right configure.h settings?

![BLT_defect](https://user-images.githubusercontent.com/16907942/57732914-79176f00-769e-11e9-85d4-1da63098ef70.jpg)

**Comment by mennovanhout** [comment]
> @mennovanhout
> ATTENTION!
> Do not use a low resistor value!
> I tried 330 Ohms to get reliable triggering and killed my BLTouch!
> Luckily only the two schottky diodes were defect and i could replace them :)
> So there is definately a problem with the configuration setup.
> The GPIO seems to be pulling low and thus we hardly can get the minimum voltage level to sense the line as high!
> Maybe some developer can help us with the right configure.h settings?
> 
> ![BLT_defect](https://user-images.githubusercontent.com/16907942/57732914-79176f00-769e-11e9-85d4-1da63098ef70.jpg)

My bltouch is working fine but have some issues with the z-offset it can't get low enough.
Nice to hear that you could fix your BLTouch! I'm a developer myself but not familiar with Marlin. I did a few tests yesterday changing some settings in configuration.h but did not yet succeed to fix the offset issue.

**Comment by ghost** [comment]
> Maybe some developer can help us with the right configure.h settings?

IF:
There is a series resistor (often this is: 1k), a pull-up resistor (often this is 10k), and a capactitor externally on the GPIO pin
AND:
You measure 5V on the signal free floating, but 2V only when connected to the controller
THEN:
Perhaps, due the configuration the GPOI pin itself is configured into "PULL DOWN" mode. It SHOULD be configured into PULL UP mode.
CHECK:
configuration.h. this section:
```
// Enable pullup for all endstops to prevent a floating state
#define ENDSTOPPULLUPS
#if DISABLED(ENDSTOPPULLUPS)
  // Disable ENDSTOPPULLUPS to set pullups individually
  //#define ENDSTOPPULLUP_XMAX
  //#define ENDSTOPPULLUP_YMAX
  //#define ENDSTOPPULLUP_ZMAX
  //#define ENDSTOPPULLUP_XMIN
  //#define ENDSTOPPULLUP_YMIN
  //#define ENDSTOPPULLUP_ZMIN
  //#define ENDSTOPPULLUP_ZMIN_PROBE
#endif

// Enable pulldown for all endstops to prevent a floating state
//#define ENDSTOPPULLDOWNS
#if DISABLED(ENDSTOPPULLDOWNS)
  // Disable ENDSTOPPULLDOWNS to set pulldowns individually
  //#define ENDSTOPPULLDOWN_XMAX
  //#define ENDSTOPPULLDOWN_YMAX
  //#define ENDSTOPPULLDOWN_ZMAX
  //#define ENDSTOPPULLDOWN_XMIN
  //#define ENDSTOPPULLDOWN_YMIN
  //#define ENDSTOPPULLDOWN_ZMIN
  //#define ENDSTOPPULLDOWN_ZMIN_PROBE
#endif
```
At least that is a place to start...

**Comment by ghost** [comment]
Concerning the capacitor:
Many users report that removing has helped them. Especially when the capacitor was 5uF or more.
100nF is not really an issue, in my opinion. ANTClabs still states: remove it.
The main issue is the current sourcing capability of the BLTouch output from the 5V / 3.3V rail to produce a logic high on the controller pin AT ALL, regardless of how long it takes to get there - unless the capacitor is so large that the voltage doesn't even reach a high enough level before the pulse is over. See the scope waveforms further up.

**Comment by ghost** [comment]
I would also implore you to mention which of @InsanityAutomation's firmwares or which other firmware you are using and in which version and to post your configuration files, at least.

**Comment by mennovanhout** [comment]
> I would also implore you to mention which of @InsanityAutomation's firmwares or which other firmware you are using and in which version and to post your configuration files, at least.

Do you know how to lower the z-offset in marlin configuration? Since the screen doesn't allow me to turn it lower.

**Comment by ghost** [comment]
#define Z_PROBE_OFFSET_FROM_EXTRUDER -1.3   // Z offset: -below +above  [the nozzle]

I always set that one, need maybe one or two iterations to get it real close. Never need to adjust Z-offset again until you change the printer somehow.

**Comment by mennovanhout** [comment]
> #define Z_PROBE_OFFSET_FROM_EXTRUDER -1.3 // Z offset: -below +above [the nozzle]
> 
> I always set that one, need maybe one or two iterations to get it real close. Never need to adjust Z-offset again until you change the printer somehow.

If i do -1 the nozzle gets higher instead of lower somehow? and +1 (1) it gets lower but not low enough. Sometimes it looks like the adjust settings during print won't use that value. But i'll run some more tests today.

**Comment by chunter1** [comment]
@FanDjango
Thank you for your support!
I'm on the "Creality_DWINTest" branche.
Here is my configuration file.
As far as i see, the pulldowns are not enabled.

[Configuration.zip](https://github.com/InsanityAutomation/Marlin/files/3182119/Configuration.zip)

**Comment by chunter1** [comment]
I checked/measured the signal path again and to me it looks like there is a 10 k pullup to 5 V and a 100 nF to GND directly connected to the Z_MIN input pin.
Between the GPIO (D3) und the input there is another 10 k in series.
So i am totally confused where that voltage clamping to 2 V should come from. :/
I assume there are no components placed on the bottom side?

![Creality3D_v2_4_mobo_Z_MIN](https://user-images.githubusercontent.com/16907942/57777631-9beb6700-7722-11e9-81e6-fb5dfed00e26.jpg)

**Comment by mennovanhout** [comment]
@chunter1 Remove c48, I've spoken with antclabs they say no mather what version you have it is always a good idea to remove c48 on cr10s pro.

**Comment by chunter1** [comment]
Is the "-Z" labeled signal on the ribbon cable connector the same as the Z_MIN signal?
If so, there is a chance that the problem is located on the interface board where the feeder is mounted.
I can't check now since i am not at home...

![cr_board_2](https://user-images.githubusercontent.com/16907942/57778758-01d8ee00-7725-11e9-85a3-868d64f65054.png)

**Comment by chunter1** [comment]
Looking at the ribbon cable breakeout board, there is the optocoupler output and the LED connected to the "-Z" / Z_MIN signal line?
However, both should not pull the line low...

![ribbon_cabe_breakout_adapter_1](https://user-images.githubusercontent.com/16907942/57779725-f8508580-7726-11e9-827f-676aa0b623a8.jpg)
![ribbon_cabe_breakout_adapter_2](https://user-images.githubusercontent.com/16907942/57779726-f8508580-7726-11e9-9d63-fe5a6b975c3d.jpg)

**Comment by InsanityAutomation** [comment]
There is another z min input on the main board. Bltouch needs to be wired there. I have not found a need to remove a capacitor on the pro. a lot going on in this thread that I didn't notice until this morning when some of the email notifications flooded through. I'll read through it shortly.

**Comment by chunter1** [comment]
@InsanityAutomation
First, thanks for the effort you put into this project! ;)
So the -Z and the Z_MIN are not connected to eachother.

**Comment by ghost** [comment]
Looks like you are on the right track

**Comment by mennovanhout** [comment]
> There is another z min input on the main board. Bltouch needs to be wired there. I have not found a need to remove a capacitor on the pro. a lot going on in this thread that I didn't notice until this morning when some of the email notifications flooded through. I'll read through it shortly.

Do you know how I can lower the nozzle while printing? For leveling everything works perfect but while printing the nozzle is to high since I use the v3 bltouch. And the LCD won't let me put it lower. maximum z offset reached?

**Comment by chunter1** [comment]
I disconnected the ribbon cable from the beakout board and well... the signal level is at 5V as it should be.
So there is definately a component on the breakout board limiting the voltage and causing the trouble.
The question is now, where the alternative Z_MIN input is?

**Comment by InsanityAutomation** [comment]
There is only one on the main board itself, just the one on the daughter board cannot be used. The board diagram above shows this, and it's right next to where the capacitor that was in question is located.

Still running around in circles at the day job, the responses during business hours over here are few and far between...

**Comment by ghost** [comment]
![grafik](https://user-images.githubusercontent.com/5990805/57799018-6492af80-774e-11e9-9f12-e36a9bf9f439.png)

**Comment by chunter1** [comment]
@FanDjango
You mean the socket or the right pin?
Because the right pin is directly connected to the 5 V supply.

I guess the alternative way is to use the Z_MAX input.

**Comment by ghost** [comment]
If I may add something: The breakout board at the end of the ribbon cable was a great convenience when you wanted to connect a 12V inductive sensor to the printer controller - on less well endowed printers you needed an optocoupler so as not to destroy your controller. Here they have added this for your convenience but it has no sense in using it for a BLTouch.

I mean of course that is the socket you need

**Comment by ghost** [comment]
One of those pins is 5V, one is GND, and one is trigger-signal in ZMIN.

**Comment by chunter1** [comment]
Correct, from left to right:   Z_MIN / GND / +5 V
But i discarded to use this pin now since it is connected in parallel to the long ribbon cable and the voltage limiting breakout board.
I'm gonna try the a real clean solution using the Z_MAX pin instead.

**Comment by ghost** [comment]
I actually think you can use ZMIN and at that point of the circuit, it will not be interfered with by the "breakout board" and ribbon cable. I prefer ZMIN, but you can also go for ZMAX, no problem. Need to configure this though

**Comment by ghost** [comment]
This is what the situation looks like, more or less:
![grafik](https://user-images.githubusercontent.com/5990805/57799893-62c9eb80-7750-11e9-94c0-2dc0b592056f.png)

**Comment by chunter1** [comment]
> This is what the situation looks like, more or less:
> ![grafik](https://user-images.githubusercontent.com/5990805/57799893-62c9eb80-7750-11e9-94c0-2dc0b592056f.png)

Additionally there is a BLUE led connected to +5V in series with a resistor.
I expected the circuit to have absolutely no influence on the signal aswell but somehow it definately gets limited when the ribbon cable is connected to the breakout board.

Now i am struggeling with configuring the Z_MAX input. :)

**Comment by chunter1** [comment]
I now connected the BLTouch to the Z_MAX input but it does not work correctly.
When i send G28 and then M48 to do the repeatability test, the BLTouch works as expected and the probing passes.
However if i select on the touchscreen to home, the BLTouch does not release it's pin to probe but keeps it retracted.
Looking at the servo signal - it is present, but it does not change.
I guess i missed something in the configuration?
Here are my changes since switching from Z_MIN to Z_MAX input:

//#define Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN

#define Z_MIN_PROBE_ENDSTOP
#define Z_MIN_PROBE_PIN 19

#define USE_ZMIN_PLUG
#define USE_ZMAX_PLUG

**Comment by ghost** [comment]
Found something on the internet where all said it worked:

You will need to connect the probe to one of the unused pins - Z-max, but make sure to read the comments in the config file - and then make a couple of changes in the configuration file. Look under [Endstop Settings]. Make sure [#define USE_XMAX_PLUG] is not commented out. Depending on your probe, either enable or disable [ #define ENDSTOPPULLUP_ZMIN_PROBE]. Then, under [Z Probe Options], comment out [#define Z_MIN_PROBE_USES_Z_MIN_ENDSTOP_PIN]. Uncomment [#define Z_MIN_PROBE_ENDSTOP] and I like to uncomment and set [#define Z_MIN_PROBE_PIN X_MAX_PIN] just so I KNOW which pin it's using. Either connect the probe to the X-max pins ore change [X_MAX_PIN] to reflect the set you are using - e.g. [Z_MAX_PIN] to use the Z-max pins. This previous step is EXTREMELY important, so make sure you set that pin correctly. Finally, under [Mesh Leveling], I want to think that you need to comment out [#define Z_SAFE_HOMING]. I don't think that that feature works with the Z probe AND the z-min used together. All of the other probe settings are the same as if the probe IS the limit switch. 

SAFE_HOMING ?

**Comment by chunter1** [comment]
Guys, thank you so much for your support!
I found the problem and it is on the breakout board.
When nothing is connected to the "Z-" connector, the optocoupler is turned ON leading to a low impedance pull down of the Z_MIN line.
That was the low impedance source i was looking for all the time. :)
**SOLUTION:**
**You must place a jumper on "Z-" connector (as shown below) to keep the optocoupler from pulling down the Z_MIN line.**
Simply using a 1 kOhm or less pullup resistor is NO VALID SOLUTION since it will probably get your diodes inside the BLTouch bunred out one day as mine did immediately (see pics above).
With this jumper in place, you don'T need any additional pullup at all.

![Zminus_line_float_jumper](https://user-images.githubusercontent.com/16907942/57806643-dc1d0a80-775f-11e9-8dd0-0f25f3aee12d.jpg)

**Comment by InsanityAutomation** [comment]
Ill need to try this on a pre-april 5th bltouch V3, i didnt pay too close attention to the daughterboard at all once I had a detected input signal! Very good observation :)

**Comment by mennovanhout** [comment]
@chunter1 good work man

So if you place a jumper there, I can remove my resistor between the white and red wire?

**Comment by kuksinskiy** [comment]
> Guys, thank you so much for your support!
> I found the problem and it is on the breakout board.
> When nothing is connected to the "Z-" connector, the optocoupler is turned ON leading to a low impedance pull down of the Z_MIN line.
> That was the low impedance source i was looking for all the time. :)
> **SOLUTION:**
> **You must place a jumper on "Z-" connector (as shown below) to keep the optocoupler from pulling down the Z_MIN line.**
> Simply using a 1 kOhm or less pullup resistor is NO VALID SOLUTION since it will probably get your diodes inside the BLTouch bunred out one day as mine did immediately (see pics above).
> With this jumper in place, you don'T need any additional pullup at all.
> 


Do I need to desolder a capacitor?

**Comment by chunter1** [comment]
> @chunter1 good work man
> 
> So if you place a jumper there, I can remove my resistor between the white and red wire?

Exactly
Did just one quick run of the probe accuracy test M48 yesterday, but the values are now perfect compared to the pullup mess before.
When i remember correctly, i have approx. 0,002 mm standard deviation now which i think is fine.

@kuksinsky
I would remove it to get the best performace possible.
However without the cap you have weaker protection and noise suppression.

Configuring the BLTouch output to be push-pull may maximize the performance since the rising edge time is no longer depending on the 10 k pullup resister alone but gets actively driven high.
To profit from this you need to use:
> #define BLTOUCH_FORCE_5V_MODE // For BLTouch V3.0 force 5V only mode

**Comment by kuksinskiy** [comment]
> > @chunter1 good work man
> > So if you place a jumper there, I can remove my resistor between the white and red wire?
> 
> Exactly
> Did just one quick run of the probe accuracy test M48 yesterday, but the values are now perfect compared to the pullup mess before.
> When i remember correctly, i have approx. 0,002 mm standard deviation now which i think is fine.
> 
> @kuksinsky
> I would remove it to get the best performace possible.
> However without the cap you have weaker protection and noise suppression.
> 
> Configuring the BLTouch output to be push-pull may maximize the performance since the rising edge time is no longer depending on the 10 k pullup resister alone but gets actively driven high.
> To profit from this you need to use:
> 
> > #define BLTOUCH_FORCE_5V_MODE // For BLTouch V3.0 force 5V only mode
Thanks for the answer! And one more question, is the correct connection for v3 in this [video](https://youtu.be/PLKEhdznVMY)?

**Comment by chunter1** [comment]
> Thanks for the answer! And one more question, is the correct connection for v3 in this [video](https://youtu.be/PLKEhdznVMY)?

Yes, however i don't know why he doesn't use the +5V and GND pin of servo output 0 in one line as they are all the same for each servo.
These are the pins i used since i want to have them all alligned and cleanly installed on one 3-pin connector :)

![bltouch_on_servo0](https://user-images.githubusercontent.com/16907942/57831760-4b254e00-77b6-11e9-85a3-5268724e21a3.png)

**Comment by chunter1** [comment]
It's a bit off topic, but i think the settings for bed size and offsets should look like this to avoid crashing into the bed holder clamps.

```
#if (ENABLED(MachineCR10Std))
#define X_BED_SIZE 300
#define Y_BED_SIZE 300
#define Z_MAX_POS 400
#endif

// Travel limits (mm) after homing, corresponding to endstop positions.
#define X_MIN_POS 0
#define Y_MIN_POS -5
#define Z_MIN_POS 0
#define X_MAX_POS X_BED_SIZE
#define Y_MAX_POS Y_BED_SIZE
```

By the way... is there something like a dedicated Creality CR-10S PRO optimization thread around that covers every setting of the printer (PID, hardware, etc.)?
I am completely new to 3d printing and would like to optimize my printer's soft- and hardware as much as possible before printing excessively. :)

**Comment by mennovanhout** [comment]
@chunter1 Are you able to print now? I'm still facing the issue that my nozzle is to high while printing.

**Comment by chunter1** [comment]
Did a quick testprint yesterday - worked well.
When the BLTouch pin is retracted, how much mm is it above your nozzle?
If i remember correctly, mine is approx. 3 mm above.

**Comment by mennovanhout** [comment]
> Did a quick testprint yesterday - worked well.
> When the BLTouch nozzle is retracted, how much mm is it above your nozzle?
> If i remember correctly, mine is approx. 3 mm above.

My nozzle seems to be 7mm to high. what really weird is when I go to movement -> z homeing and then lower the z so it is 0 the nozzle is perfect. But while printing in the adjust settings the z offset is -1.52 and is still way too high? And I can't lower it the LCD won't let me.

**Comment by chunter1** [comment]
I followed the mechanical installation guide here:
https://docs.wixstatic.com/ugd/f5a1c8_5a3d7fd6382f4ff5a77fa65b7f4028b9.pdf

**Comment by mennovanhout** [comment]
> I followed the mechanical installation guide here:
> https://docs.wixstatic.com/ugd/f5a1c8_5a3d7fd6382f4ff5a77fa65b7f4028b9.pdf

Yo uare using 1.1.9B6 right from this creality_dwinset branch can you share your changes? But not the changes about the probe pin only the settings of z offset etc?

**Comment by chunter1** [comment]
> > I followed the mechanical installation guide here:
> > https://docs.wixstatic.com/ugd/f5a1c8_5a3d7fd6382f4ff5a77fa65b7f4028b9.pdf
> 
> Yo uare using 1.1.9B6 right from this creality_dwinset branch can you share your changes? But not the changes about the probe pin only the settings of z offset etc?

I am using the "Creality_DWINTest" branche.
However... since i am completely new to 3d-printing and Marlin, it is still not clear to me what branches are suitable for the CR-10S PRO and which ones are recommended. (maybe someone can give some advice/overview?)
Even I'm not sure if the InsanityAutomation repo is the right way to go or use plain Marlin??
Still in the learning process... ;)

These are the settings i remember - i can post the others later today:

```
X_PROBE_OFFSET_FROM_EXTRUDER -27
Y_PROBE_OFFSET_FROM_EXTRUDER -1
Z_PROBE_OFFSET_FROM_EXTRUDER -2.1
```

**Comment by ghost** [comment]
> I am using the "Creality_DWINTest" branche.

@InsanityAutomation has put a lot of effort into putting together firmware(s) for CR printers. **At the current moment, the Creality_DWINTest is suitable for the BLTouch V3.0 but issues a number of unneeded commands, making the probing slightly slower than necessary. It contains a lot of presets and stuff specially made for CR printers.** In the near future he will be updating his branch(es) with the newest and latest stuff re. BLTOUCH to get up to date. It's based on 1.9. I definitely recommend until you get more adventurous.

"Plain" Marlin is fine if you want to do all this configuration stuff and tweaking all by yourself or want to add stuff that goes beyond the standard. In that case, I would recommend bugfix-2.0.x, though.

CReality also offer firmwares, but I have not ever looked at them, so I cannot say what they are based on.

**Comment by mennovanhout** [comment]
> > > I followed the mechanical installation guide here:
> > > https://docs.wixstatic.com/ugd/f5a1c8_5a3d7fd6382f4ff5a77fa65b7f4028b9.pdf
> > 
> > 
> > Yo uare using 1.1.9B6 right from this creality_dwinset branch can you share your changes? But not the changes about the probe pin only the settings of z offset etc?
> 
> I am using the "Creality_DWINTest" branche.
> However... since i am completely new to 3d-printing and Marlin, it is still not clear to me what branches are suitable for the CR-10S PRO and which ones are recommended. (maybe someone can give some advice/overview?)
> Even I'm not sure if the InsanityAutomation repo is the right way to go or use plain Marlin??
> Still in the learning process... ;)
> 
> These are the settings i remember - i can post the others later today:
> 
> ```
> X_PROBE_OFFSET_FROM_EXTRUDER -27
> Y_PROBE_OFFSET_FROM_EXTRUDER -1
> Z_PROBE_OFFSET_FROM_EXTRUDER -2.1
> ```

Thanks could you also check which lines you've set these changes.

**Comment by chunter1** [comment]
> > > > I followed the mechanical installation guide here:
> > > > https://docs.wixstatic.com/ugd/f5a1c8_5a3d7fd6382f4ff5a77fa65b7f4028b9.pdf
> > > 
> > > 
> > > Yo uare using 1.1.9B6 right from this creality_dwinset branch can you share your changes? But not the changes about the probe pin only the settings of z offset etc?
> > 
> > 
> > I am using the "Creality_DWINTest" branche.
> > However... since i am completely new to 3d-printing and Marlin, it is still not clear to me what branches are suitable for the CR-10S PRO and which ones are recommended. (maybe someone can give some advice/overview?)
> > Even I'm not sure if the InsanityAutomation repo is the right way to go or use plain Marlin??
> > Still in the learning process... ;)
> > These are the settings i remember - i can post the others later today:
> > ```
> > X_PROBE_OFFSET_FROM_EXTRUDER -27
> > Y_PROBE_OFFSET_FROM_EXTRUDER -1
> > Z_PROBE_OFFSET_FROM_EXTRUDER -2.1
> > ```
> 
> Thanks could you also check which lines you've set these changes.

It's around line 1322 in the section beginning with:
`#if (ENABLED(ABL_BLTOUCH) && ENABLED(HotendStock))`

**Comment by juliandroid** [comment]
@chunter1 Great findings! I was looking for a long time to understand the core problem!

> > These are the settings i remember - i can post the others later today:
> > ```
> > X_PROBE_OFFSET_FROM_EXTRUDER -27
> > Y_PROBE_OFFSET_FROM_EXTRUDER -1
> > Z_PROBE_OFFSET_FROM_EXTRUDER -2.1
> > ```

Quick question - what is the mount adapter/bracket that you have used to mount the BLTouch? Obviously your offsets are related to that bracket?

**Comment by chunter1** [comment]
> @chunter1 Great findings! I was looking for a long time to understand the core problem!
> 
> > > These are the settings i remember - i can post the others later today:
> > > ```
> > > X_PROBE_OFFSET_FROM_EXTRUDER -27
> > > Y_PROBE_OFFSET_FROM_EXTRUDER -1
> > > Z_PROBE_OFFSET_FROM_EXTRUDER -2.1
> > > ```
> 
> Quick question - what is the mount adapter/bracket that you have used to mount the BLTouch? Obviously your offsets are related to that bracket?

This one here:
https://www.thingiverse.com/thing:3454594

**Comment by mennovanhout** [comment]
> > > > > I followed the mechanical installation guide here:
> > > > > https://docs.wixstatic.com/ugd/f5a1c8_5a3d7fd6382f4ff5a77fa65b7f4028b9.pdf
> > > > 
> > > > 
> > > > Yo uare using 1.1.9B6 right from this creality_dwinset branch can you share your changes? But not the changes about the probe pin only the settings of z offset etc?
> > > 
> > > 
> > > I am using the "Creality_DWINTest" branche.
> > > However... since i am completely new to 3d-printing and Marlin, it is still not clear to me what branches are suitable for the CR-10S PRO and which ones are recommended. (maybe someone can give some advice/overview?)
> > > Even I'm not sure if the InsanityAutomation repo is the right way to go or use plain Marlin??
> > > Still in the learning process... ;)
> > > These are the settings i remember - i can post the others later today:
> > > ```
> > > X_PROBE_OFFSET_FROM_EXTRUDER -27
> > > Y_PROBE_OFFSET_FROM_EXTRUDER -1
> > > Z_PROBE_OFFSET_FROM_EXTRUDER -2.1
> > > ```
> > 
> > 
> > Thanks could you also check which lines you've set these changes.
> 
> It's around line 1322 in the section beginning with:
> `#if (ENABLED(ABL_BLTOUCH) && ENABLED(HotendStock))`

You know that if statement isn't where he gets it from?

![probe offset](https://user-images.githubusercontent.com/26040589/57883112-99256a80-7825-11e9-8595-d742c4c2447a.png)
our printer****

**Comment by ghost** [comment]
Guys, you do know that you can measure / estimate these values yourself? If X and Y offset is wrong by a couple of mm, it's not a heartbreak.
And if Z is (say) too large by a millimeter and the first layer is in the air, lower it by a millimeter and try again.
Just set X and Y by using a small ruler and your eyes, and iteratively get Z offset down to the right value. Especially as BLTOUCH V3.0 and newer has slightly longer strokes, so I doubt that these values for Z offset are totally right anymore.

**Comment by mennovanhout** [comment]
> Guys, you do know that you can measure / estimate these values yourself? If X and Y offset is wrong by a couple of mm, it's not a heartbreak.
> And if Z is (say) too large by a millimeter and the first layer is in the air, lower it by a millimeter and try again.
> Just set X and Y by using a small ruler and your eyes, and iteratively get Z offset down to the right value. Especially as BLTOUCH V3.0 and newer has slightly longer strokes, so I doubt that these values for Z offset are totally right anymore.

My problem is more that it seems that my cr10s pro doesn't react to the z prob setting. I see exactly no diffrence. As you can see I even tested it with -7. That should defintly hit the bed and be way lower then 0. But it isn't?

**Comment by ghost** [comment]
It means: Try a Z offset value and cancel the print after the first couple of cm's of printing and correct the value, recompile, flash, and try again. OR LCD MOVE AXIS down until the paper is just fixed under the nozzle and subtract the distance you went down from the Z offset, recompile etc. etc. Takes an hour max.

**Comment by ghost** [comment]
Ah, ok. What about G500, G503. THis is in the EEPROM!

**Comment by ghost** [comment]
You can compile this and flash it until you are grey in the face, it will take the old value stored in eeprom

**Comment by ghost** [comment]
Many things are in the eeprom, so: Such stuff alway check in the LCD configuration menus if they have successfully been used or the eeprom values are still in effect. Or do the **M502 M500** sorry earlier post was wrong

**Comment by ghost** [comment]
M502 Factory reset <- use compiled values
M500 Store <- store these in eeprom

**Comment by mennovanhout** [comment]
> M502

going to test that right now thanks!

**Comment by ghost** [comment]
But you must also put them into the eeprom with M500

**Comment by mennovanhout** [comment]
> But you must also put them into the eeprom with M500

I did! I changes values run your 2 comments and going to do a test print now. Hopefully I see a diffrence and then I can fine tune it.

**Comment by ghost** [comment]
It also depends all a little on what the GCODE at the start of your print is doing, of course. I always do lot's of stuff, but main thing in my start code is M502, then M500, G28, then G29, then print. Nothing else... but not everyone wants to level before each print, ok?

**Comment by mennovanhout** [comment]
OMG it works FINALLY You are my hero!

**Comment by ghost** [comment]
Oof. Happy now

**Comment by kuksinskiy** [comment]
> OMG it works FINALLY You are my hero!

Could you share your final version?

**Comment by mennovanhout** [comment]
> > OMG it works FINALLY You are my hero!
> 
> Could you share your final version?

No issue at al! I'll be making a video this evening and post it here if I do not forget with how exactly I got my pre 5th april BLTouch to work.

**Comment by chunter1** [comment]
I just calibrated the extruder and the default settings are spot on.
Did anybody already do an extruder and bed PID autotune?
Wonder if your results are similar to mine:

```
Hotend:
#define DEFAULT_Kp 31.26
#define DEFAULT_Ki 2.77
#define DEFAULT_Kd 88.35

Bed:
#define DEFAULT_bedKp 274.71
#define DEFAULT_bedKi 38.54
#define DEFAULT_bedKd 489.46
```

**Comment by InsanityAutomation** [comment]
@chunter1 ive confirmed every version of bltouch functions with no resistor with that jumper in place, Pre and Post april 5th V3.0 and both revs V3.1 ive been sent! Im still a bit surprised none of us noticed that sooner! I havnt pulled that board out of mine lol Good spotting there, im sure youll make a ton of people pretty happy here!

**Comment by ghost** [comment]
What's the best way to spread this? Info to ANTClabs?

**Comment by juliandroid** [comment]
I'm wondering why BLTouch 2.x is working after going through the same optocoupler?

**Comment by ghost** [comment]
More power to drive the led in the opto? V3. 1 maybe can do it better who knows

**Comment by InsanityAutomation** [comment]
Only the original v3.0 didn't have enough muscle in the output to overcome it. Every other version has. Short run of probes, but still a ton out there!

**Comment by Goldbacher** [comment]
My BLTouch V3 works on motherboard 2.4 without jumper. Should I still attach the jumper?

**Comment by mennovanhout** [comment]
> It also depends all a little on what the GCODE at the start of your print is doing, of course. I always do lot's of stuff, but main thing in my start code is M502, then M500, G28, then G29, then print. Nothing else... but not everyone wants to level before each print, ok?

Can I have your start code? It worked 1 time for me now my nozzle is way to high again.

**Comment by ghost** [comment]
> Can I have your start code? It worked 1 time for me now my nozzle is way to high again.

How can that be?

But ok, here's my current code for one of my printers:

```
; Printer Begin GCODE
; Note: Would be useful: {speed_travel} is mm/s, but need mm/min

M502 ; Set to Marlin defaults
M500 ; Store in EEPROM

M107 ; Fan off
M84 E; Extruder off

G91   ; Relative

M140 S{material_bed_temperature_layer_0} ; nowait

M104 S150 ; nowait

G1 Z+25 F1800 ; up a little for some safety before home XY
G28 X Y              ; Home XY at 50mm + unknown
G28 Z                 ; Home Z no. 1 and then...

G90 ; Absolute

G1 Z180 F1800  ; go up quickly nearly all the way
M211 S0             ; turn off software endstops
G1 Z190 F120    ; and up slowly into physical limiter because of ANET A8 dual Z unsync danger
M211 S1             ; turn on software endstops
G1 Z15 F1800    ; and down again quickly nearly all the way and then
G28 Z                 ; home Z again, now both Z motors are in sync

G34

G1 X3 Y{machine_depth} Z10 F10000; Present bed for cleaning

M190 S{material_bed_temperature_layer_0} ;wait
G29
M420 S1 Z5    ; set Leveling fade height to 5 mm

M104 S{material_print_temperature_layer_0, 0} ;nowait

; Present nozzle for cleaning while heating
G1 X3 Y3 Z150 F10000 ; X=3 Y=3 inner corner

M109 S{material_print_temperature_layer_0, 0} ;wait
M190 S{material_bed_temperature_layer_0} ;wait

M83 ; E relative
G92 ﻿E0.0                               ; reset extrusion distance
G1 Z0.3                F1800       ; close to bed
G1 X30      E+6.0 F1000.0    ; 20mm intro line @ 0.30
G1 X70      E+1.5 F1000.0    ; +40mm intro line @ 0.08
G1 X110    E+3.0 F1000.0    ; +40mm intro line ﻿@ 0.15
G1             E-1.0  F900          ; retract to avoid stringing
G1 X116    E0      F1000.0     ; +6mm intro line @ 0.00
G1 X111    E0      F1000.0     ; -5mm intro line @ 0.00﻿
G1 Z10                F1800        ; Up to 1cm
G92 E0.0                               ; reset extrusion distance

M84 S600     ; steppers off if inactive longer than 10 minutes (safety)

M900 K0.2    ; Linear Advance Coefficient (experimental)
; turn off coasting in Cura !!!
; or set this M900 value to 0.0

M82 ; E absolute (currently, this is absolutely expected by CURA
```

**Comment by ghost** [comment]
> My BLTouch V3 works on motherboard 2.4 without jumper. Should I still attach the jumper?

If it works, I think I would not at this moment. Are you using a resistor?

**Comment by Goldbacher** [comment]
Nothing, I have installed the jumper now, also working. I think, it is the newer BLTouch V3. Ordered from china at April 11. I use the V-G-S-Connector D11 and not AIO for V-G.

**Comment by ghost** [comment]
Yeah, sounds like a After-Apr 5 one. If you really wanted to, you could analyse that QR code but I wouldn't trouble myself

**Comment by juliandroid** [comment]
Unless it is something older on stock.
@Goldbacher type the serial number on the right of the QRcode at: http://www.id2db.com/genuine10.php

**Comment by Goldbacher** [comment]
Date of manufacture: 2019-04-20.

**Comment by quarky42** [comment]
> > @chunter1
> Configuring the BLTouch output to be push-pull may maximize the performance since the rising edge time is no longer depending on the 10 k pullup resister alone but gets actively driven high.
> To profit from this you need to use:
> 
> > #define BLTOUCH_FORCE_5V_MODE // For BLTouch V3.0 force 5V only mode

Has this been tested?  I'm considering it for a new BLTouch V3.0 install, but am curious if there is any more intonation about this and did it work well?  Any dangers or issues I should know before trying this particular setting?

**Comment by chunter1** [comment]
> > > @chunter1
> > > Configuring the BLTouch output to be push-pull may maximize the performance since the rising edge time is no longer depending on the 10 k pullup resister alone but gets actively driven high.
> > > To profit from this you need to use:
> > 
> > 
> > > #define BLTOUCH_FORCE_5V_MODE // For BLTouch V3.0 force 5V only mode
> 
> Has this been tested? I'm considering it for a new BLTouch V3.0 install, but am curious if there is any more intonation about this and did it work well? Any dangers or issues I should know before trying this particular setting?

I have my printer running well with this setup since installation.
Verified the signal levels with the scope and they are perfect.

**Comment by InsanityAutomation** [comment]
@quarky42 Thats been the definitive mode for all creality machines for some time! Keep in mind this will change a bit shortly...

**Comment by quarky42** [comment]
> ive confirmed every version of bltouch functions with no resistor with that jumper in place, Pre and Post april 5th V3.0 and both revs V3.1 ive been sent! 

@InsanityAutomation - 
Thank you!  Just got a chance to check and my v3.0 is from March 28th.   

@chunter1 thank you very much!  I look forward to trying that.  Making some progress at getting everything reassembled.  (BLTouch, all metal hot end, jumper installed, fan upgrades mostly complete and touch screen updated.  Need to reflash the firmware after adding that 5V mod.

@InsanityAutomation thanks.  I'm piecing information together and trying to get a better but picture view.  With all the variations (hardware and firmware) and with all the discussions, it has been a bit of a challenge.  I am still very thankful for your hard work and information as well as all the contributors!  I'm just saying it's been tricky for me trying to get smarter on this stuff.  I have some programming experience so it isn't completely foreign to me, but jumping in the middle of it has been tricky.

**Comment by quarky42** [comment]
@InsanityAutomation can you elaborate on your last comment about it changing?  Maybe link me to a post / thread that I could see what you're referring to specifically?  Or is care to explain?

**Comment by InsanityAutomation** [comment]
Hopefully we are nearing the end of the road on everything changing all over the place... V3.1 probes start shipping soon and we have some more changes to support those coming shortly. With any luck thatll close the saga....

 See recent commits on this branch

https://github.com/InsanityAutomation/Marlin/tree/BLTouchV31Followups

**Comment by quarky42** [comment]
@InsanityAutomation  Is the Creality_DWINTest still preferable right now?  I just got everything reassembled and will be tinkering with the arduino project / related settings tonight.

**Comment by InsanityAutomation** [comment]
Yup, it's stable. Other dwin branch is still missing features.

**Comment by quarky42** [comment]
I think I understood you correctly.  The BLTouchV31Followups and the other BLTouch (V3Matchup and V3Tweaks) branches are not prime time, right?  You just gave me the link to the followups to see some of the current development going on, I think.

**Comment by MrPaulAR** [comment]
I'm having issues with CR10s Pro and BLTouch v3 (Mfg date: 2019-03-13) running  CR10SPro_BLTouch_B6_115k.hex.  I haven't removed the capacitor yet nor have I installed a resistor.  Before doing one of those physical mods I want to confirm it is or isn't needed as it's not quite clear to me.  I've also tried compiling my own firmware with the exact same results so I rolled back to the hex version.

If I install it without the jumper the probe extends & retracts when I touch it but the Z keeps going down.  This seems to be the issue that everyone has above.

If I install the jumper however I get an immediate trigger and the Z axis never goes down.  The PIN extends, retracts, then says it's homed.  Video @ https://photos.app.goo.gl/uSUUTc2iMjcCqNLx6

**Comment by InsanityAutomation** [comment]
@MrPaulAR Install the jumper as chunter1 noted above! Best method with no modification and eliminates what caused the need for a pullup resistor!

**Comment by quarky42** [comment]
@MrPaulAR  I have a V3 built at the end of March and am not having the problems you describe.   Can you make sure that your white wire of the 2 pin wires is towards the middle of the board on the Z- End Stop Connector (on the control board) and the black wire is on the middle pin of that connector?

I did not remove any capacitors or add any resistors and mine is working.

Did you edit the Configuration.h?    Looks like the current version of that file on the GitHub has some things on it that you might want to check:
Line 59:  "#define HotendAllMetal"    Comment that out.  Not part of your problem, but unless you have an all metal hot end, then you should probably not have that enabled.
Line 97: "//#define ABL_BLTOUCH"   That needs to be uncommented.  Remove the // in the front of it.

(Edit: I stand corrected, removing the lines that aren't needed.)

If Line 97 was commented out then your BLTouch wasn't enabled.

You also might need to update the lines 1163, 1164 and 1165 with your BLTouch offset as those defaults might not be accurate for how you have it installed.

**Comment by InsanityAutomation** [comment]
These are unnecessary : 

Line 105: "//#define ABL_UBL" That needs to be uncommented, I believe. If someone corrects me, go with what they say. I'm not an expert. Just trying to help you out.
Line 120: "//#define MeshFine" Consider uncommenting that one and commenting out the line above it. Not critical, but just my 2 cents.
Line 517: //#define AUTO_POWER_CONTROL This might need to be uncommented. This might explain the problems I'm having with this firmware right now. (I'll try this tonight and see what happens unless someone more experienced chimes in and can confirm one way or the other.)

But flipping black / white wires is absolutely the most common issue I see!

**Comment by quarky42** [comment]
Thank you, @InsanityAutomation     Can I please get some help with my print temperature problem?   I created a new issue/ new thread on it.   If you saw it early on, I've revised it a number of times and it might be worth a fresh look to see what I've added to it and tried to explain better what is happening.

Edit: I see your reply now.  Thank you for looking at it.  I'm sorry I didn't see the reply at first.

**Comment by MrPaulAR** [comment]
Thanks @InsanityAutomation and @quarky42.  I'm just an idiot and had it plugged into Y_MAX instead of Z_MIN.  Not sure what I was doing when I plugged it in but it defiantly wasn't right.  Moved the cable to the right spot and it's now stopping z home when I touch the bottom of the sensor.

**Comment by quarky42** [comment]
> 
> 
> Thanks @InsanityAutomation and @quarky42. I [...] just [...] had it plugged into Y_MAX instead of Z_MIN. 

I'm really glad it was an easy fix!  Now do me a favor:  When you're ready do a small test print and let me know how it goes.   I'm trying to chase down my own problems.

If you have a serial connection to the printer via USB and can print your M503 (and save a copy of that for yourself), I'd be interested in your M301 values that are displayed there.   If you wanted to take it a step further you could manually run M303 E0 S200 C10     (200 is 200 degrees C and that test is being ran 10 times...at the end of it, in the serial terminal shell it will tell you what it measured for your P I D settings and I'd be very curious what values you get.  That command by itself won't make any changes to your settings, it will only tell us what it calculates your values to be.  I want to see how different they are from the defaults you have loaded at this point.

**Comment by MrPaulAR** [comment]
I printed off an XYZ cube and it went fine.  Pretty much the same as my Ender 3 before it got struck by lightning (Path == ethernet > switch > computer > switch > pi > serial port > ender 3).  New mainboard has been ordered.  TMI but whatever.

Here is your request @quarky42 
M301 P20.84 I1.96 D55.47

PID Autotune finished! Put the last Kp, Ki and Kd constants from below into Configuration.h
#define DEFAULT_Kp 26.36
#define DEFAULT_Ki 2.10
#define DEFAULT_Kd 82.60

**Comment by quarky42** [comment]
Wow regarding the lightning, but damn sorry to hear about it.  Crazy to have it surge through so many devices.  

Thank you very much for the info.  Very helpful.  My D value is quite different than yours.   I'm glad to have that!

**Comment by chunter1** [comment]
> I printed off an XYZ cube and it went fine. Pretty much the same as my Ender 3 before it got struck by lightning (Path == ethernet > switch > computer > switch > pi > serial port > ender 3). New mainboard has been ordered. TMI but whatever.
> 
> Here is your request @quarky42
> M301 P20.84 I1.96 D55.47
> 
> PID Autotune finished! Put the last Kp, Ki and Kd constants from below into Configuration.h
> #define DEFAULT_Kp 26.36
> #define DEFAULT_Ki 2.10
> #define DEFAULT_Kd 82.60

Interesting... mine look quite different:

```
Hotend:
#define DEFAULT_Kp 31.26
#define DEFAULT_Ki 2.77
#define DEFAULT_Kd 88.35

Bed:
#define DEFAULT_bedKp 274.71
#define DEFAULT_bedKi 38.54
#define DEFAULT_bedKd 489.46
```

**Comment by quarky42** [comment]
My gut feeling from looking at many runs and trying 5 cycles, 10  cycles, and even 20 cycles is that those look proportionally similar.  So even though the values are different, their combined effect are along the same lines.   

On a side note, I found out that my hot end loses temperature in cycles even the heated bed is on.  If I turn the heated bed off, my hot end is stable.

**Comment by juliandroid** [comment]
Lol, Antclabs took the credit for finding the solution with the jumper: 

`We immediately start to solve the issue, and we found the reason of the issue and also simple solution.`

https://www.antclabs.com/creality3d-board?fbclid=IwAR3QdEJKMqMwluMPNzmAhOiQ5pq6zgGtXtF8PIXY_Ekb4hyMKL4NLKkJjRE

I'm disgusted :/

**Comment by InsanityAutomation** [comment]
Gah! I've mentioned it and who found it on Facebook, discord and reddit! Even though I am a bit embarrassed for having not noticed it myself far earlier....

**Comment by quarky42** [comment]
I mentioned to BLTOUCH about how this solution was being talked about far before them.  They updated their link to acknowledge that.  Not a bad step on their part to try and give credit where it is due.

**Comment by InsanityAutomation** [comment]
> 
> 
> I mentioned to BLTOUCH about how this solution was being talked about far before them. They updated their link to acknowledge that. Not a bad step on their part to try and give credit where it is due.

They had actually sent Tinymachines an email with an acknowledgement of that awhile ago as well. They say they were not aware that chunter had found it here before they did. Given how quickly they came around to acknowledging it, im inclined to take their word for it.

As I think everything is known and all the info here is fairly spread, im going to close the issue though.

**Comment by snekya** [comment]
endless thanks for this post.  
power cycling my machine was resulting in the sensor doing everything correctly but triggering the board to stop Z.  
for about 3 weeks my solution was leaving my machine on after a firmware update... maybe the firmware update was bypassing this optocoupler function somehow?  Anyways, this post just saved me so much frustration and time.   I no longer have to leave my machine on LOL
